Remove global state from play.api.mvc.Security

gmethvin opened the pull request:
I don't believe this helper is used that often, since typically we recommend a third-party library for handling authentication/authorization, so I've just deprecated the code that reads from configuration.

marcospereira commented on framework/src/play/src/main/scala/play/api/mvc/Security.scala:
```diff
@@ -59,8 +67,17 @@ object Security {
+    Play.privateMaybeApplication.flatMap(_.configuration.getOptional[String]("session.username")) match {
+      case Some(usernameKey) =>
+        logger.warn("The session.username configuration key is no longer supported.")
+        logger.warn("You can remove this warning by manually reading the ")
```
"reading the "...

There is something missing here. :-)

gmethvin replied:
thanks

richdougherty commented:
LGTM. This API isn't mentioned anywhere in our docs is it?

richdougherty commented:
CI test timed out on `play.api.libs.TemporaryFileReaperSpec`, restarting the build.

marcospereira commented:
> This API isn't mentioned anywhere in our docs is it?

It is not. But we should update the Migration26.md page to add information about this deprecation, just as we did for the Java version.

gmethvin commented:
@marcospereira Actually it turns out it's used in the cache docs: https://www.playframework.com/documentation/2.5.x/ScalaCache#Caching-HTTP-responses

Anyway, I updated the example and added information to the migration guide.
